torch/fx/graph_module.py

```diff
@@ -704,12 +704,15 @@ def __reduce__(self):
     # we need to define deepcopy otherwise it will call __reduce__
     # and cause symbolic tracing to occur every time we try to copy the object
     def __deepcopy__(self, memo):
+        res = GraphModule(self, self.graph)
+        memo[id(self)] = res
         fake_mod = torch.nn.Module()
-        fake_mod.__dict__ = copy.deepcopy(self.__dict__)
-        res = GraphModule(fake_mod, fake_mod.__dict__['_graph'])
-        res.meta = copy.deepcopy(getattr(self, 'meta', {}))
+        fake_mod.__dict__ = copy.deepcopy(self.__dict__, memo)
+        GraphModule.__init__(res, fake_mod, fake_mod.__dict__['_graph'])
```

Comment:
should we be calling this constructor? does it work if GraphModule is subclassed?

Reply:
that subclass need to also override `__deepcopy__` if it has additional data.